src/GraphQL/Extensions/ObjectExtensions.cs

```diff
@@ -28,175 +27,230 @@ public static class ObjectExtensions
             }
 
-            foreach (var item in source)
+            return new ReflectionInfo
+            {
+                Constructor = bestConstructor,
+                CtorFields = ctorFields,
+                MemberFields = members,
+            };
+
+            static (MemberInfo, bool) FindMatchingMember(
+                [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.PublicFields)]
+                Type type,
+                string propertyName)
             {
-                // these parameters have already been used in the constructor, no need to set property
-                if (usedKeys?.Any(k => k == item.Key) == true)
-                    continue;
-
-                string propertyName = GetPropertyName(item.Key, out var field);
                 PropertyInfo? propertyInfo = null;
 
+                // note: analzyer raises false IL2070 warning due to BindingFlags.IgnoreCase being present
+
                 try
                 {
+#pragma warning disable IL2070 // 'this' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The parameter of method does not have matching annotations.
                     propertyInfo = type.GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
+#pragma warning restore IL2070 // 'this' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The parameter of method does not have matching annotations.
                 }
                 catch (AmbiguousMatchException)
                 {
+#pragma warning disable IL2070 // 'this' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The parameter of method does not have matching annotations.
                     propertyInfo = type.GetProperty(propertyName, BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+#pragma warning restore IL2070 // 'this' argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The parameter of method does not have matching annotations.
                 }
 
-                if (propertyInfo != null && propertyInfo.CanWrite)
+                if (propertyInfo?.SetMethod?.IsPublic ?? false)
                 {
-                    object? value = GetPropertyValue(item.Value, propertyInfo.PropertyType, field.ResolvedType
-                        ?? throw new InvalidOperationException($"Could not get ResolvedType for field '{field.Name}' of type '{inputGraphType}'."));
-                    propertyInfo.SetValue(obj, value, null); //issue: this works even if propertyInfo is ValueType and value is null
+                    var isExternalInit = propertyInfo.SetMethod.ReturnParameter.GetRequiredCustomModifiers()
+                        .Any(type => type.FullName == "System.Runtime.CompilerServices.IsExternalInit");
```

Comment:
This is painfully slow for some reason, slowing the whole function by 50%+.  I cached the results of FindMatchingMember to mitigate against this.  So now all reflection calls are cached within `_types` or `_members` (besides the reflection calls to construct the type and calls to `SetValue`).